### DISCUSSION TITLE
Roles from Users from an other Organisation

### DISCUSSION THREAD

**Discussion by aheissenberger** [zitadel/zitadel#7560] [Q&A] [answered]
My Setup:
* ORG 01
  * Project 01 - Settings: Assert Roles on Auth
    * Roles
      * Admin
      * Approved
    * Grants
      * ORG 02
    * Users
      * User 01  (roles: Admin)
    * Apps
      * App 01 - Token Settings: User roles inside ID Token
 * ORG 02
   * Users
      * User 02 (roles: Approved)

I have a ReactJS App based on the example from documentation.

`User 01` from `ORG 01` will include the role `Admin`
`User 02` from  `ORG 02` will not include any roles info

And how can I fix the problem with the missing roles from users from other orgs and how can I add the information what org the user belongs and meta infos from the org and user to the `userinfo_endpoint` result? Where in the docs is a descripton of the claims to request the informations.

**Comment by hifabienne** [comment, marked as answer]
hei @aheissenberger 
Generally you have the docuemantation about different scopes here:
Standard Scopes: https://zitadel.com/docs/apis/openidoauth/scopes#standard-scopes
Reserved Scopes (ZITADEL specific) https://zitadel.com/docs/apis/openidoauth/scopes#reserved-scopes

In the reserved scope s you have the resource owner scope: urn:zitadel:iam:user:resourceowner this will include the organization of the user, not sure what you mean with meta infos but with the metadata scope you can include the metadata (key values) from the user urn:zitadel:iam:user:metadata

@livio-a @muhlemmer can you help with the first question, about how to include the roles in any case?

**Reply by aheissenberger** [reply]
I still have the problem that the user info api will not include the roles of the user when the user is from an other organization. Any ideas what needs to be done to get access to this data?

**Reply by livio-a** [reply]
I just tried to reproduce, but could not find an issue. I also authenticated with a user, which has an authorization on a granted project (as the setup above) with the following scopes: `openid email profile urn:zitadel:iam:user:resourceowner`
I get back the info (resourceowner and roles) in the tokens and userinfo_endpoint.
Do you encountered this issue on zitadel.cloud or self-hosted? If the latter, which version do you currently use?

**Reply by aheissenberger** [reply]
I have to confess, that it does work and might be a provision error on my side with the user. I might have been added the role to a user which had a role with the "new" Button - see #4952.

I do use a self-hosted version of Zitadel (v2.44.1) but upgraded everything today before I tested again today.
I also replicated my setup with a cloud test account and it worked there too.
